src/components/table/AppTable.vue

```diff
@@ -82,7 +82,7 @@
   </table>
 </template>
 
-<script lang="ts" setup>
+<script lang="ts" setup generic="I extends Item">
```

Comment:
Interesting feature, how would I pass a generic parameter to this component?

Reply:
As in when you use the component? I don't know how you explicitly pass types but for the most part it will infer them from your props

Reply:
good to know, I'll try it out when I get the chance, thanks